neuralop/layers/segment_csr.py

```diff
@@ -54,38 +60,23 @@ def segment_csr(
         # if batched, shape [b, n_reps, channels]
         # otherwise shape [n_reps, channels]
         if src.ndim == 3:
-            batched = True
             point_dim = 1
         else:
-            batched = False
             point_dim = 0
 
         # if batched, shape [b, n_out, channels]
         # otherwise shape [n_out, channels]
         output_shape = list(src.shape)
-        n_out = indptr.shape[point_dim] - 1
+        n_out = indptr.shape[0] - 1
+        n_reps = indptr[1:] - indptr[:-1]
+
         output_shape[point_dim] = n_out
 
-        out = torch.zeros(output_shape, device=src.device)
+        indices = torch.arange(n_out).to(indptr.device).repeat_interleave(n_reps)
+        out = torch.zeros(output_shape, device=src.device).index_add_(point_dim, indices, src)
```

Comment:
We might want to add some more comments here or maybe break in two to make these clearer? Since the indices are fixed (1..n_out), perhaps we can do this without the repeat interleaves - e.g with broadcasting, need to think about it more.

Reply:
We could use broadcasting via a mask but that would add memory overhead.